src/test/java/edu/kis/powp/jobs2d/events/TestIfCommandFits.java

```diff
@@ -0,0 +1,46 @@
+package edu.kis.powp.jobs2d.events;
+
+import edu.kis.powp.jobs2d.PaperFormats;
+import edu.kis.powp.jobs2d.command.CompoundCommand;
+import edu.kis.powp.jobs2d.command.DriverCommand;
+import edu.kis.powp.jobs2d.command.OperateToCommand;
+import edu.kis.powp.jobs2d.command.SetPositionCommand;
+import edu.kis.powp.jobs2d.features.CommandsFeature;
+import edu.kis.powp.jobs2d.features.FeatureManager;
+import edu.kis.powp.jobs2d.visitor.ExceededCanvasVisitor;
+
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+import static org.junit.Assert.assertTrue;
+
+public class TestIfCommandFits implements ActionListener {
+
+	private Shape bounds;
+	private final static Logger logger = Logger.getLogger(Logger.GLOBAL_LOGGER_NAME);
+
+	public TestIfCommandFits(Shape bounds) {
+		this.bounds = bounds;
+	}
+
+	@Override
+	public void actionPerformed(ActionEvent actionEvent) {
+		CommandsFeature commands = FeatureManager.getFeature(CommandsFeature.class);
+		CompoundCommand complex = (CompoundCommand)commands.getDriverCommandManager().getCurrentCommand();
+
+		ExceededCanvasVisitor visitor = new ExceededCanvasVisitor(PaperFormats.getA4());
+		visitor.visit(complex);
```

Comment:
You use the visitor incorrectly. Please use accept method (then casting to CompoundCommand won't be necessary)